CaloFilters/src/CaloClusterCounterFilter_module.cc

```diff
@@ -17,134 +17,164 @@
+  virtual bool filter(art::Event& event) override;
+  virtual bool beginRun(art::Run& run);
+  virtual bool endRun(art::Run& run) override;
+
+  using Parameters = art::EDFilter::Table<Config>;
+  explicit CaloClusterCounter(const Parameters& conf);
+
+private:
+  int _diagLevel;
+  int _nProcess;
+  int _nPass;
+  art::InputTag _clTag;
+  double _minClEnergy;
+  double _maxClEnergy;
+  double _minClRadius;
+  int _minNCl;
+  std::vector<int> _diskID;
+  int _minNCel;
+  int _maxNCel;
+
+  const Calorimeter* _calogeom;
+};
+
+CaloClusterCounter::CaloClusterCounter(const Parameters& config) :
+    art::EDFilter{config}, _diagLevel(config().diag()), _nProcess(0), _nPass(0),
+    _clTag(config().CCTag()), _minClEnergy(config().minClE()), _maxClEnergy(config().maxClE()),
+    _minClRadius(config().minClR()), _minNCl(config().minNCl()), _diskID(config().diskID()),
+    _minNCel(config().minNCel()), _maxNCel(config().maxNCel()) {
+
+  produces<TriggerInfo>();
+}
+
+void CaloClusterCounter::beginJob() {}
+
+void CaloClusterCounter::endJob() {}
+
+bool CaloClusterCounter::beginRun(art::Run& run) {
+  GeomHandle<Calorimeter> ch;
+  _calogeom = ch.get();
+
+  return true;
+}
+
+bool CaloClusterCounter::endRun(art::Run& run) {
+  if (_diagLevel > 0 && _nProcess > 0) {
+    std::cout << "[CaloClusterCounter::endRun]"
+              << " passed " << _nPass << " events out of " << _nProcess << " for a ratio of "
+              << float(_nPass) / float(_nProcess) << std::endl;
   }
+  return true;
+}
+
+//--------------------------------------------------------------------------------
+// Follow the body of the Filter logic
+//--------------------------------------------------------------------------------
+bool CaloClusterCounter::filter(art::Event& event) {
+
+  ++_nProcess;
+  if (_nProcess % 10 == 0 && _diagLevel > 0)
+    std::cout << "[CaloClusterCounter::filter] Processing event from CaloClusterCounter =  "
+              << _nProcess << std::endl;
+
+  std::unique_ptr<TriggerInfo> triginfo(new TriggerInfo);
+  bool retval(false);
+
+  // Get calo cluster collection
+  auto clH = event.getValidHandle<CaloClusterCollection>(_clTag);
+  const CaloClusterCollection* caloClusters = clH.product();
+
+  int nClusterAboveThreshold(0);
+
+  // for loop over the clusters in the calorimeter
+  for (auto icl = caloClusters->begin(); icl != caloClusters->end(); ++icl) {
+    auto const& cluster = *icl;
+    CLHEP::Hep3Vector pos = cluster.cog3Vector();
+    float clRadius = std::sqrt(std::pow(pos.x(), 2) + std::pow(pos.y(), 2));
+    if (_diagLevel > 0) {
+      std::cout << "[CaloClusterCounter::filter] cluster "
+                << std::distance(caloClusters->begin(), icl) << " diskID = " << cluster.diskID()
+                << " E = " << cluster.energyDep() << " radial distance = " << clRadius
+                << " nCels = " << cluster.size() << std::endl;
+    }
+    if ((std::find(_diskID.begin(), _diskID.end(), cluster.diskID()) != _diskID.end()) &&
```

Comment:
I'm confused why you use std::find here, since the conditional you apply seems trivial.  Or is it possible for clusters to have a non-physical diskID?  If so, how are such clusters created?

Reply:
I wanted to make the module more flexible for other possible use. I wanted to enable the possibility of using both disks; _diskID is the data-member with the allowed diskIDs and in the RMC specific config is set to `[0]`, while to use both disks it should be set to `[0, 1]`